### DISCUSSION TITLE
Restoring the latest screen after exiting the application

### DISCUSSION THREAD

**Discussion by onelenyk** [arkivanov/Decompose#498] [Q&A]
Hello!
I'm facing the next behavior:

after removing the app from active apps, and opening that app again, the app restores the latest screen that was in the stack,  and the screen becomes unable to interact with

How to exclude that behavior?
I don't need to restore the app when it was killed by the user.

**Comment by arkivanov** [comment]
Thanks for the question! I believe you definitely want this behaviour. This is standard Android [state preservation](https://developer.android.com/topic/libraries/architecture/saving-states). E.g. when you open a camera app to take a photo, or your app is interrupted by a phone call, the system may kill your app in the background. Then when the user goes back to your app, it should be in the same state as before (e.g. the taken photo should be processed). I recommend making sure your app preserves all required state.

If you really want to disable state preservation, you can try setting `persistent = false` in your root `Child Stack`. See the [docs](https://arkivanov.github.io/Decompose/navigation/stack/overview/#screen-recreation-and-process-death-on-not-only-android) . Let me know if for some reason this doesn't work for your case. There are other ways as well.

**Comment by onelenyk** [comment]
Thank you for such a quick answer! 
Yes, you are completely right about the cases you described, but how to properly close the application, when the user wanted to kill it? 

How can I set up my component or child stack, to properly react to user intent to close an app?

I also was trying to use persistent = false, in every of my childStack, not only in the root, but it still continues to recreate the latest screen after opening the app.

I have this dependencies:
    implementation "com.arkivanov.decompose:decompose:2.0.1"
    implementation "com.arkivanov.decompose:extensions-compose-jetpack:2.0.1"

**Reply by arkivanov** [reply]
Android manages the saved state on its own, Decompose doesn't do anything special here. I.e. you either receive `savedInstanceState` argument in `Activity#onCreate` or not. When the user swipes out your app from the Recents screen, perhaps the state is preserved if there is a Service running in background. But I'm not sure.

**Comment by onelenyk** [comment]
I have noticed that my application did not become inactive after killing by swiping out. 
But when I stopped my application through the notifications panel, after restarting it started from the entry point, as I expected.

So I think that is a side effect from my Service that should work in the background 

upd: not sure anymore about that, more testing shows different behaviours.

**Comment by onelenyk** [comment]
We have fixed it, the problem was because of we used Koin to inject the Root Component.

Thanks for your time!
